workflow/rocoto/tasks.py

```diff
@@ -39,15 +38,16 @@ class Tasks:
 
     def __init__(self, app_config: AppConfig, run: str) -> None:
 
-        self.app_config = copy.deepcopy(app_config)
+        self.app_config = app_config
         self.run = run
-        # Re-source the configs with RUN specified
-        print(f"Source configs with RUN={run}")
-        self._configs = self.app_config.source_configs(run=run, log=False)
+
+        # Get the configs for the specified RUN
+        self._configs = self.app_config.configs[run]
 
         # Update the base config for the application
-        self._configs['base'] = self.app_config.update_base(self._configs['base'])
-        # Save dict_configs and base in the internal state (never know where it may be needed)
+        self._configs['base'] = self.app_config._update_base(self._configs['base'])
```

Comment:
Is there a reason the limited function of `_update_base` can't be handled at init time instead of deferring it to this point?

Reply:
I'm going to test this out, but I think that since `_update_base` is defined in the subclass, this needs to be in `_init_finalize` or copied into each of the subclass `__init__` methods.

Reply:
The `_update_base` isn't an issue.  However, the subclass `__init__` method (at least the `GFSCycledAppConfig` class) sets some flags needed by the subclass's `_get_app_configs` method.  I could resolve that by moving these flag declarations to the `AppConfig` `__init__` method (e.g. `do_jediatmvar`, `do_jediocnvar`, `do_hybvar`), but that seems like a kludge.  I prefer the `_init_finalize` method as it is.